substrate/bin/utils/chain-spec-builder/src/lib.rs

```diff
@@ -310,3 +329,21 @@ pub fn generate_chain_spec_for_runtime(cmd: &CreateCmd) -> Result<String, String
 		(false, false) => chain_spec.as_json(false),
 	}
 }
+
+/// Chain-spec extension that provides access to optional `relay_chain` and `para_id` fields.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, ChainSpecGroup, ChainSpecExtension)]
+pub struct OptionalParaFieldsExtension {
```

Comment:
Why do we need this here?

Reply:
Removed. If you are asking why we need it at all, without any extension `relay_chain` and `para_id` fields are silently removed from the chain specs.

Reply:
Then we should change this to pass through any unknown fields and just modify the stuff we know.

Reply:
Okay, modified this PR to just allow adding substitutes